### ISSUE TITLE
Docker credentials fails for secrets created with kubectl v1.13.0

### ISSUE THREAD

**Issue by squaremo** [weaveworks/flux#1596]
If you create a dockerconfigjson secret for use as an imagePullSecret, using kubectl v.1.13.0, fluxd is not able to parse it when it comes to scanning the image registry.

The reason is that the format generated by v1.13.0 is different to that prior: using v.1.12,

```
$ kubectl create secret docker-registry docker-reg-secret --docker-server=private.dockerrepo.com --docker-username=xxxxx --docker-password="xxxxx" --docker-email="xxx@xyz.cim" --dry-run -o json | jq -r '.data[".dockerconfigjson"]' | base64 -d
{"auths":{"private.dockerrepo.com":{"username":"xxxxx","password":"xxxxx","email":"xxx@xyz.cim","auth":"eHh4eHg6eHh4eHg="}}}
```

Using v1.13.0:

```
$ kubectl create secret docker-registry docker-reg-secret --docker-server=private.dockerrepo.com --docker-username=xxxxx --docker-password="xxxxx" --docker-email="xxx@xyz.cim" --dry-run -o json | jq -r '.data[".dockerconfigjson"]' | base64 -d
{"auths":{"private.dockerrepo.com":{"Username":"xxxxx","Password":"xxxxx","Email":"xxx@xyz.cim"}}}
```

It's the lack of an `auth` field that trips fluxd up; but the capitalisation might also cause a problem. It's unclear whether this was an entirely deliberate change; we can _probably_ work around it by looking for `Username` and `Password` fields (in preference to decoding `auth`, even).

**Comment by squaremo** [comment]
@awh tracked this down to https://github.com/kubernetes/kubernetes/commit/9f5c2aea1a398b7afa045d28fa35aeaa1612bd18, which redefines `DockerConfigEntry` without the Auth field and without the JSON struct annotations.

**Comment by dananichev** [comment]
So... any ETA on fixing this?

**Comment by squaremo** [comment]
Looks like it's been fixed in kubectl, in time for 1.14: https://github.com/kubernetes/kubernetes/pull/72344

I am inclined to add an entry to troubleshooting.md advising people to create secrets with a kubectl either side of 1.13.

**Comment by dananichev** [comment]
@squaremo thanks, updated `kubectl` to `1.13.2` on the client side and generated manifests with secrets.

**Comment by squaremo** [comment]
@dananichev Oh, so it's fixed in 1.13.2? Hurrah!

**Comment by ryanhartje** [comment]
This doesn't seem to be the case, following this issue, this is still ending in the same failure mode for me when generating the secret with both 1.13.2 and 1.14. 

In my case fluxd is built with kubectl 1.14 as well, and I'm still getting:
```
ts=2019-05-04T17:29:10.903152477Z caller=images.go:148 component=cluster resource=repo:deployment/repo err="decoded credential has wrong number of fields (expected 2, got 1)"
```
